src/server.rs

```diff
@@ -381,11 +387,13 @@ async fn stream_classification_with_gen(
 
 async fn detection_content(
     State(state): State<Arc<ServerState>>,
+    headers: HeaderMap,
     Json(request): Json<models::TextContentDetectionHttpRequest>,
 ) -> Result<impl IntoResponse, Error> {
     let request_id = Uuid::new_v4();
     request.validate()?;
-    let task = TextContentDetectionTask::new(request_id, request);
+    let filtered_headers = filter_header(&state.orchestrator.get_config(), headers).await;
+    let task = TextContentDetectionTask::new(request_id, request, filtered_headers);
```

Comment:
Update per other changes.

```suggestion
    let headers = filter_headers(&state.orchestrator.config().passthrough_headers, headers);
    let task = TextContentDetectionTask::new(request_id, request, headers);
```